ui/src/app/job-details/resources/resources.component.ts

```diff
@@ -59,12 +59,17 @@ export class JobResourcesComponent implements OnInit {
       }
 
       if (this.job.extensions.logs) {
-        for (let file of Object.keys(this.job.extensions.logs)) {
-          this.readResourceFile(file);
-          let tabId = 'log-' + file;
-          this.tabIds.push(tabId);
-          this.tabTitles.set(tabId, file);
-        }
+        let files = Object.keys(this.job.extensions.logs);
+        Promise.all(files.map(file => this.readResourceFile(file)))
```

Comment:
Actually, rereading the `Promise.all` docs, you don't exactly want this as it will short-circuit reject. You'd need to catch immediately after `this.readResourceFile`, `handleError` there, then `Promise.all` on the return value there. You'd then need to filter out "empty" results below.

Reply:
Done.